stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewHighlightOverlays.java

```diff
@@ -0,0 +1,269 @@
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+      return new ViewHighlightOverlaysJellybeanMR2();
+    }
+    return new NoOpViewHighlightOverlays();
+  }
+
+  private static class NoOpViewHighlightOverlays extends ViewHighlightOverlays {
+
+    @Override
+    void highlightView(View view, int mainColor) {
+    }
+
+    @Override
+    void removeHighlight(View view) {
+    }
+  }
+
+  @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
+  private static class ViewHighlightOverlaysJellybeanMR2 extends ViewHighlightOverlays {
+    private static final int MARGIN_OVERLAY_COLOR = 0xaaf4ca9e;
+    private static final int PADDING_OVERLAY_COLOR = 0xaabedab6;
+    private final HighlightDrawable[] mHighlightDrawables = {
+        new MainHighlightDrawable(),
+        new PaddingTopHighlightDrawable(),
+        new PaddingBottomHighlightDrawable(),
+        new PaddingRightHighlightDrawable(),
+        new PaddingLeftHighlightDrawable(),
+        new MarginTopHighlightDrawable(),
+        new MarginBottomHighlightDrawable(),
+        new MarginRightHighlightDrawable(),
+        new MarginLeftHighlightDrawable()
+    };
+
+    ViewHighlightOverlaysJellybeanMR2() {
+    }
+
+    @Override
+    void highlightView(View view, int mainColor) {
+      mHighlightDrawables[0].setColor(mainColor);
+
+      int total = mHighlightDrawables.length;
+      for (int i = 0; i < total; i++) {
+        mHighlightDrawables[i].highlightView(view);
+      }
+    }
+
+    @Override
+    void removeHighlight(View view) {
+      for (ColorDrawable drawable : mHighlightDrawables) {
+        view.getOverlay().remove(drawable);
+      }
+    }
+
+    abstract class HighlightDrawable extends ColorDrawable {
+
+      protected final Rect mMargins = new Rect();
+      protected final Rect mPaddings = new Rect();
+
+      HighlightDrawable(int color) {
+        super(color);
+      }
+
+      public HighlightDrawable() {
+      }
+
+      void highlightView(View view) {
+        ViewGroup.LayoutParams layoutParams = view.getLayoutParams();
+        if (layoutParams instanceof MarginLayoutParams) {
+          MarginLayoutParams marginLayoutParams = (MarginLayoutParams) layoutParams;
+          mMargins.left = marginLayoutParams.leftMargin;
+          mMargins.top = marginLayoutParams.topMargin;
+          mMargins.right = marginLayoutParams.rightMargin;
+          mMargins.bottom = marginLayoutParams.bottomMargin;
+        }
+        mPaddings.left = view.getPaddingLeft();
+        mPaddings.top = view.getPaddingTop();
+        mPaddings.right = view.getPaddingRight();
+        mPaddings.bottom = view.getPaddingBottom();
+      }
+    }
+
+    class MainHighlightDrawable extends HighlightDrawable {
+
+      @Override
+      void highlightView(View view) {
+        super.highlightView(view);
+        setBounds(0, 0, view.getWidth(), view.getHeight());
+        view.getOverlay().add(this);
```

Comment:
This line of code, `view.getOverlay().add(this);` is repeated in every single HighlightDrawable. For symmetry with `ViewHighlightOverlaysJellybeanMR2.removeHighlight()`, why not just put this into the `ViewHighlightOverlaysJellybeanMR2.highlightView()` method?